src/transformers/pipelines/image_to_text.py

```diff
@@ -69,76 +71,37 @@ def _sanitize_parameters(self, max_new_tokens=None, generate_kwargs=None):
                 - An image loaded in PIL directly
-
                 The pipeline accepts either a single image or a batch of images.
-            texts (`str`, `List[str]`, *optional*):
-                The pipeline handles conditional generation. If `texts` is provided, the model will generate the image
-                caption conditioned on the text.
-
             max_new_tokens (`int`, *optional*):
                 The amount of maximum tokens to generate. By default it will use `generate` default.
-
-            kwargs (`Dict`, *optional*):
-                The kwargs will contain the preprocessing kwargs (including kwargs related to text input) as well as
-                the generate kwargs that will send all of these arguments directly to `generate` allowing full control
-                of this function.
-
+            generate_kwargs (`Dict`, *optional*):
+                Pass it to send all of these arguments directly to `generate` allowing full control of this function.
         Return:
             A list or a list of list of `dict`: Each result comes as a dictionary with the following key:
-
             - **generated_text** (`str`) -- The generated text.
         """
-        model_inputs = None
-        if isinstance(images, list) and texts is not None and isinstance(texts, list):
-            model_inputs = [{"images": image, "texts": texts} for image, texts in zip(images, texts)]
-        elif isinstance(images, list) and texts is not None:
-            # same text on all images
-            model_inputs = [{"images": image, "texts": texts} for image in images]
-        elif isinstance(images, list) and texts is None:
-            # no text
-            model_inputs = [{"images": image} for image in images]
-        elif not isinstance(images, list) and texts is None:
-            # classic input with only images
-            model_inputs = images
-        elif not isinstance(images, list) and texts is not None:
-            # classic input with images and text
-            model_inputs = {"images": images, "texts": texts}
-
-        if model_inputs is None:
-            raise ValueError("Input is not valid - got {} and {} for image and text".format(images, texts))
-
-        return super().__call__(model_inputs, **kwargs)
-
-    def preprocess(self, model_input):
-        images = model_input["images"] if isinstance(model_input, dict) else model_input
-        if isinstance(images, list):
-            images = [load_image(image) for image in images]
-        else:
-            images = load_image(images)
+        return super().__call__(images, **kwargs)
 
-        if isinstance(model_input, dict) and "texts" in model_input:
-            texts = model_input["texts"]
-        else:
-            texts = None
+    def preprocess(self, images, texts=None):
+        images = load_image(images)
+
+        if texts is not None and isinstance(texts, list) and len(texts) > 1:
```

Comment:
Call it just text (without s) and just check it's a `str` ?

In terms of names `text` is not highly explicit in terms of intent. 

- `prompt`
- `preprompt`
- `output_prefix` 
?

If None are better, feel free to ignore.

Reply:
I would say `conditional_prompt` would be nice!

Sounds very good for the first check, let's only check if it's `str`